gumby/scenario.py

```diff
@@ -145,47 +145,47 @@ def _parse_arguments(self, args):
 
         return unnamed_args, named_args
 
-    def _parse_for_loop(self, loop):
-        """
-        Parse a for loop, checking for its syntactical and lexical correctness and returning its functional components
-
-        :param loop: a string containing the for loop. The string should have the following structure:
-                     'for <control_variable> in <lower_bound> to <higher_bound> call <callable> <optional_parameters>
-                     <peer_specification>'. The value of the lower_bound needn't actually be lower than the higher
-                     bound.
-        :return: return a tuple containing the following: the callable, the callable's argument line (unparsed),
-                 the lower_bound, the higher_bound, an offset (either 1 of -1) indicating the for's index in/decrements,
-                 the control variable (including the '$' character at the beginning)
-        """
-        parts = loop.split(' ', 7)
-
-        # Less than 7 tokens means that we had a malformed 'for'
-        if len(parts) < 7:
-            raise Exception()
-
-        offset = 1
-        control_var, lo_bound, hi_bound, callable_function = parts[::2]
-        args = parts[7] if len(parts) == 8 else ''
-
-        # Check if the other tokens are correct, so as to avoid ambiguity
-        if ['in', 'to', 'call'] != [x.lower() for x in parts[1:6:2]]:
-            raise Exception()
-
-        # Convert the bound strings to integers
-        lo_bound = int(lo_bound)
-        hi_bound = int(hi_bound)
-
-        # control_var
-        control_var = '$' + control_var
-
-        # The lower and upper bounds are considered inclusive, hence the additional in/decrements
-        if lo_bound > hi_bound:
-            offset = -1
-            hi_bound -= 1
-        else:
-            hi_bound += 1
-
-        return callable_function, args, lo_bound, hi_bound, offset, control_var
+##    def _parse_for_loop(self, loop):
```

Comment:
Instead of commenting out the code, please remove it 👍

Reply:
Thanks Martijn, I will do that asap

Reply:
In addition, there are also some tests related to for-loops. You can also remove them.

Reply:
Only 1 for-loops related test has been detected and removed ( file: gumby/tests/scenario_language )